lib/GADS/Datum/File.pm

```diff
@@ -153,10 +153,30 @@ sub _build_files
     elsif ($self->has_ids) {
         @return = $self->_ids_to_files(@{$self->ids});
     }
+    
+    return [+{
+        id => -1,
+        name => 'Purged',
+        mimetype => 'text/plain',
+    }] if $self->is_purged;
```

Comment:
Move this above the previous block, otherwise the previous block will be processed but do nothing